### PR TITLE
Do a softer subscription check

### LINKED ISSUE #3296: Subscription checking to hard
The subscription check for certain plugins will fail as soon as it is exceeded by only one token.

If a plugin with 5000 subscriptions is hit by a system with 20.000 users, the subscription check should be fine for 5000 cases out of 20.000.

### PR DESCRIPTION
**Pull request by cornelinux**
Subscription check should not fail for the subscribed users.

Closes #3296

### CODE REVIEW

File: tests/test_lib_subscriptions.py [hunk 1]
```diff
@@ -103,9 +103,14 @@ def test_03_check_subscription(self):
 
         init_token({"type": "spass"}, user=User("shadow", self.realm1))
         init_token({"type": "spass"}, user=User("nopw", self.realm1))
-        # Now we have three users with tokens, subscription will fail
-        self.assertRaises(SubscriptionError, check_subscription,
-                          "demo_application")
+        # Now we have three users with tokens, but only two are allowed. We fail with a probabiliy of 1/3
```
Comment (plettich) [inline tests/test_lib_subscriptions.py:106]: I think this is bound to produce randomly failing tests. We should mock out the randomness for the test and just check if we get the required error.

**Review by plettich — CHANGES_REQUESTED** [review]

File: tests/test_lib_subscriptions.py (hunk 1 above)
Comment (cornelinux) [reply tests/test_lib_subscriptions.py:107]: done in ae1b513d

**Comment by plettich** [comment]
Now the tests fail with a different `random` error :-) I'll take a look.

**Review by plettich — APPROVED** [review]

**Comment by plettich** [comment]
> Now the tests fail with a different `random` error :-) I'll take a look.

This will be fixed with #3303

**Pull request merged**